tests/test_schemas.py

```diff
@@ -172,22 +216,274 @@ def test_sub_schema_graph(self):
+        reordered_nodes = nodes.copy()
+        reordered_nodes.remove('Patient')
+        reordered_nodes.append('Patient')
+        assert reordered_nodes != expected_nodes
+
+    @pytest.mark.parametrize("data_model", list(DATA_MODEL_DICT.keys()), ids=list(DATA_MODEL_DICT.values()))
+    def test_gather_all_nodes(self, helpers, data_model):
+        # Instantiate Parser
+        data_model_parser = helpers.get_data_model_parser(data_model_name=data_model)
+
+        # Parse Model
+        attr_rel_dictionary = data_model_parser.parse_model()
+
+        # Instantiate DataModelNodes
+        data_model_nodes = generate_data_model_nodes(helpers, data_model_name=data_model)
+
+        all_nodes = data_model_nodes.gather_all_nodes(attr_rel_dict=attr_rel_dictionary)
+
+        # Make sure there are no repeat nodes
+        assert len(all_nodes) == len(set(all_nodes))
+
+        # Check that nodes from first entry, are recoreded in order in all_nodes
+        # Only check first entry, bc subsequent ones might be in the same order as would be gathered with gather_nodes if it contained a node that was already recorded.
+        first_attribute = list(attr_rel_dictionary.keys())[0]
+        attr_info = (first_attribute, attr_rel_dictionary[first_attribute])
+        expected_starter_nodes = data_model_nodes.gather_nodes(attr_info=attr_info)
+        actual_starter_nodes = all_nodes[0:len(expected_starter_nodes)]
+
+        assert actual_starter_nodes == expected_starter_nodes
+
+    def test_get_rel_node_dict_info(self, helpers, relationships):
+        # Instantiate Parser
+        data_model_parser = helpers.get_data_model_parser(data_model_name='example.model.csv')
+
+        # Instantiate DataModelNodes
+        data_model_nodes = generate_data_model_nodes(helpers, data_model_name='example.model.csv')
+
+        for relationship in relationships:
+            rel_dict_info = data_model_nodes.get_rel_node_dict_info(relationship)
+            if rel_dict_info:
+                assert type(rel_dict_info[0]) == str
+                assert type(rel_dict_info[1]) == dict
+                assert 'default' in rel_dict_info[1].keys()
+
+    @pytest.mark.parametrize("data_model", list(DATA_MODEL_DICT.keys()), ids=list(DATA_MODEL_DICT.values()))
+    def test_get_data_model_properties(self, helpers, data_model):
+        # Instantiate Parser
+        data_model_parser = helpers.get_data_model_parser(data_model_name=data_model)
+
+        # Parse Model
+        attr_rel_dictionary = data_model_parser.parse_model()
+
+        # Instantiate DataModelNodes
+        data_model_nodes = generate_data_model_nodes(helpers, data_model_name=data_model)
+
+        # Get properties in the data model
+        data_model_properties = data_model_nodes.get_data_model_properties(attr_rel_dictionary)
+        
+        # In the current example model, there are no properties, would need to update this section if properties are added.
+        assert data_model_properties == []
```

Comment:
Yeah, this is a good point. I created a ticket here to track: https://sagebionetworks.jira.com/jira/software/c/projects/FDS/issues/FDS-1226